src/ralph/data_center/models/virtual.py

```diff
@@ -26,13 +27,28 @@ def __str__(self):
         return 'Database: {}'.format(self.service_env)
 
 
+class VIPProtocol(Choices):
+    _ = Choices.Choice
+
+    TCP = _('TCP')
+    UDP = _('UDP')
+
+
 class VIP(AdminAbsoluteUrlMixin, BaseObject):
+    name = models.CharField(_('name'), max_length=255)
+    ip = models.ForeignKey(IPAddress)
+    port = models.PositiveIntegerField(verbose_name=_('port'), default=0)
+    protocol = models.PositiveIntegerField(
+        verbose_name=_('protocol'), choices=VIPProtocol(),
+    )
+
     class Meta:
         verbose_name = _('VIP')
         verbose_name_plural = _('VIPs')
+        unique_together = ('ip', 'port', 'protocol')
 
     def __str__(self):
-        return 'VIP: {}'.format(self.service_env)
+        return self.name
```

Comment:
@mkurek - I'm just wondering, how VIPs should be represented, i.e. is `self.name` enough for that, or maybe we should go with "full" representation (IP + port + protocol), since names alone are not unique here..?